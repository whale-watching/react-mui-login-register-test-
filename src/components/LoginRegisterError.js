import React, {Component} from 'react';
import {withStyles} from '@material-ui/core/styles';
import PropTypes from 'prop-types';

const styles = theme => ({
  root: {
    color: theme.palette.error.main,
    fontWeight: 'bold',
    margin: [[theme.spacing.unit, 0]]
  }
});

class LoginRegisterError extends Component {
  static propTypes = {
    message: PropTypes.string.isRequired
  };

  render() {
    return <div className={this.props.classes.root}>{this.props.message}</div>
  }
}

export default withStyles(styles)(LoginRegisterError);