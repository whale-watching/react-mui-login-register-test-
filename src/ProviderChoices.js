import React, {Component} from 'react';
import {withStyles} from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import {FacebookButton, GithubButton, TwitterButton} from "./components/ProviderButtons";
import {
  PROVIDER_FACEBOOK,
  PROVIDER_GITHUB,
  PROVIDER_TWITTER
} from "./"

const styles = theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    "& > *+*": {
      marginTop: theme.spacing.unit
    }
  }
});

class ProviderChoices extends Component {
  static propTypes = {
    login: PropTypes.bool,
    register: PropTypes.bool,
    onChoice: PropTypes.func,
    providers: PropTypes.arrayOf(PropTypes.string)
  };

  render() {
    const {
      classes,
      login,
      register,
      providers
    } = this.props;

    const commonProps = {
      login,
      register,
      variant: 'raised',
      color: 'primary',
      className: classes.button,
      onClick: this.handleClick
    };

    return (
        <div className={classes.root}>
          {providers.includes(PROVIDER_FACEBOOK) && <FacebookButton {...commonProps}/>}
          {providers.includes(PROVIDER_GITHUB) && <GithubButton {...commonProps}/>}
          {providers.includes(PROVIDER_TWITTER) && <TwitterButton {...commonProps}/>}
        </div>
    );
  }

  handleClick = providerId => {
    if (this.props.onChoice) {
      this.props.onChoice(providerId);
    }
  };
}

export default withStyles(styles)(ProviderChoices);