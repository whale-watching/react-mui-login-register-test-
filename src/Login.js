import React, {Component, Fragment} from 'react';
import {withStyles} from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import ProviderChoices from "./ProviderChoices";
import LocalUserLogin from "./LocalUserLogin";
import _ from 'lodash';

const styles = theme => ({
  root: {
    "& > * + *": {
      marginTop: theme.spacing.unit * 2
    }
  },
  or: {
    textAlign: 'center'
  }
});

class Login extends Component {
  static propTypes = {
    onLogin: PropTypes.func,
    onLoginWithProvider: PropTypes.func,
    providers: PropTypes.arrayOf(PropTypes.string),
    loginFailed: PropTypes.string,
    disableLocal: PropTypes.bool
  };

  render() {
    const {
      classes,
      onLogin,
      onLoginWithProvider,
      loginFailed,
      providers,
      disableLocal
    } = this.props;
    return (
        <div className={classes.root}>
          {
            !disableLocal &&
            <LocalUserLogin onLogin={onLogin} loginFailed={loginFailed}/>
          }

          {
            !disableLocal && !_.isEmpty(providers) &&
            <div className={classes.or}>or</div>
          }

          {!_.isEmpty(providers) &&
            <ProviderChoices login
                             onChoice={onLoginWithProvider}
                             providers={providers}
            />
          }
        </div>
    );
  }
}

export default withStyles(styles)(Login);
