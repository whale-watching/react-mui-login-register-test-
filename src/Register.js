import React, {Component, Fragment} from 'react';
import {withStyles} from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import _ from 'lodash';

import ProviderChoices from "./ProviderChoices";
import LocalUserRegister from "./LocalUserRegister";

const styles = theme => ({
  root: {
    "& > * + *": {
      marginTop: theme.spacing.unit * 2
    }
  },
  or: {
    textAlign: 'center'
  }
});

class Register extends Component {
  static propTypes = {
    onRegister: PropTypes.func,
    onRegisterWithProvider: PropTypes.func,
    providers: PropTypes.arrayOf(PropTypes.string),
    registerFailed: PropTypes.string,
    disableLocal: PropTypes.bool,
    disableRegisterProviders: PropTypes.bool
  };

  render() {
    const {
      classes,
      onRegister,
      onRegisterWithProvider,
      registerFailed,
      providers,
      disableLocal,
      disableRegisterProviders
    } = this.props;

    const showProviders = !_.isEmpty(providers)
      && !disableRegisterProviders;
    const showOr = !disableLocal && showProviders;

    return (
        <div className={classes.root}>
          {
            !disableLocal &&
            <LocalUserRegister onRegister={onRegister} registerFailed={registerFailed}/>
          }

          {
            showOr &&
            <div className={classes.or}>or</div>
          }

          {
            showProviders &&
            <ProviderChoices register
                             onChoice={onRegisterWithProvider}
                             providers={providers}
            />
          }
        </div>
    );
  }
}

export default withStyles(styles)(Register);
