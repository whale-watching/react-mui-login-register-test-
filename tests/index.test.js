import React from 'react';
import TestRenderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {configure, mount} from 'enzyme';

import ReactMuiLoginRegister, {
  PROVIDER_TWITTER, PROVIDER_GITHUB, PROVIDER_FACEBOOK
} from '../src';
import ProviderChoices from '../src/ProviderChoices';
import LoginRegisterError from '../src/components/LoginRegisterError';
import Login from '../src/Login';
import Register from '../src/Register';

test('render default', () => {
  const component = TestRenderer.create(
      <ReactMuiLoginRegister transitionTimeout={0}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('expected default provider buttons', () => {
  const rendered = TestRenderer.create(
      <ReactMuiLoginRegister transitionTimeout={0}/>,
  );

  const testInstance = rendered.root;

  const providerChoices = testInstance.findByType(ProviderChoices);
  const providerButtons = providerChoices.findAll(instance => instance.type.name === "ProviderButton");
  expect(providerButtons).toHaveLength(3);

  const providers = providerButtons.map(b => b.props.provider);
  expect(providers).toEqual(expect.arrayContaining([PROVIDER_GITHUB, PROVIDER_FACEBOOK, PROVIDER_TWITTER]));
});

test('expected with given providers', () => {
  const rendered = TestRenderer.create(
      <ReactMuiLoginRegister transitionTimeout={0}
                             providers={[PROVIDER_FACEBOOK, PROVIDER_GITHUB]}
      />,
  );

  const testInstance = rendered.root;

  const providerChoices = testInstance.findByType(ProviderChoices);
  const providerButtons = providerChoices.findAll(instance => instance.type.name === "ProviderButton");
  expect(providerButtons).toHaveLength(2);

  const providers = providerButtons.map(b => b.props.provider);
  expect(providers).toEqual(expect.arrayContaining([PROVIDER_GITHUB, PROVIDER_FACEBOOK]));
});

test('with a login error', () => {
  const rendered = TestRenderer.create(
      <ReactMuiLoginRegister transitionTimeout={0} loginFailed='Login error message'/>,
  );

  expect(rendered.root.findByType(Login)
  .findByType(LoginRegisterError).props.message).toEqual('Login error message');
});

describe('dom testing', () => {
  beforeAll(() => {
    configure({adapter: new Adapter()});
  });

  test('with a register error', () => {

    const rendered = mount(
        <ReactMuiLoginRegister transitionTimeout={0} registerFailed='Register error message'/>,
    );

    rendered.find('Tab').find('[label="Register"]').simulate('click');

    const errorInstance = rendered.find(Register).find(LoginRegisterError);

    expect(errorInstance).toHaveLength(1);
    expect(errorInstance.props().message).toEqual('Register error message');
  });

  test('with default, local enabled', () => {

    const rendered = mount(
      <ReactMuiLoginRegister transitionTimeout={0}/>,
    );

    expect(rendered.find('LocalUserLogin')).toHaveLength(1);
  });

  test('with local disabled', () => {

    const rendered = mount(
        <ReactMuiLoginRegister transitionTimeout={0} disableLocal={true}/>,
    );

    expect(rendered.find('LocalUserLogin')).toHaveLength(0);
  });

  test('with default, register enabled', () => {

    const rendered = mount(
      <ReactMuiLoginRegister transitionTimeout={0}/>,
    );

    const registerTab = rendered.find('Tab').find('[label="Register"]');
    expect(registerTab).toHaveLength(1);
    const tabs = rendered.find('Tabs');
    expect(tabs).toHaveLength(1);
  });

  test('with register disabled', () => {

    const rendered = mount(
        <ReactMuiLoginRegister transitionTimeout={0} disableRegister={true}/>,
    );

    const registerTab = rendered.find('Tab').find('[label="Register"]');
    expect(registerTab).toHaveLength(0);
    const tabs = rendered.find('Tabs');
    expect(tabs).toHaveLength(0);
  });
});

