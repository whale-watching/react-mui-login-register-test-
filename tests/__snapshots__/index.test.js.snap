// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default 1`] = `
<div
  className="LoginRegister-root-1"
>
  <div
    className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 MuiCard-root-3 LoginRegister-card-2"
  >
    <div
      className="MuiTabs-root-31"
    >
      <div
        className="MuiTabs-flexContainer-32"
      >
        <div
          className="MuiTabs-scroller-33 MuiTabs-fixed-34"
          onScroll={[Function]}
          role="tablist"
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <div
            className="MuiTabs-flexContainer-32"
          >
            <button
              aria-selected={true}
              className="MuiButtonBase-root-52 MuiTab-root-40 MuiTab-textColorPrimary-43 MuiTab-selected-45 MuiTab-fullWidth-47"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiTab-wrapper-48"
              >
                <span
                  className="MuiTab-labelContainer-49"
                >
                  <span
                    className="MuiTab-label-50"
                  >
                    Login
                  </span>
                </span>
              </span>
              <span
                className="MuiTouchRipple-root-58"
              />
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root-52 MuiTab-root-40 MuiTab-textColorPrimary-43 MuiTab-fullWidth-47"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiTab-wrapper-48"
              >
                <span
                  className="MuiTab-labelContainer-49"
                >
                  <span
                    className="MuiTab-label-50"
                  >
                    Register
                  </span>
                </span>
              </span>
              <span
                className="MuiTouchRipple-root-58"
              />
            </button>
          </div>
          <span
            className="TabIndicator-root-55 TabIndicator-colorPrimary-56 MuiTabs-indicator-39"
            style={
              Object {
                "left": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="TabContent-root-65"
    >
      <div
        className="Login-root-66"
      >
        <div
          className="LocalUserLogin-root-68"
        >
          <form
            className="LocalUserLogin-form-69"
            disabled={false}
            onError={[Function]}
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <div
              className="MuiFormControl-root-72 LocalUserLogin-field-70"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <label
                className="MuiFormLabel-root-81 MuiInputLabel-root-76 MuiInputLabel-formControl-77 MuiInputLabel-animated-80"
                data-shrink={false}
              >
                Username
                <span
                  className="MuiFormLabel-asterisk-85"
                >
                   *
                </span>
              </label>
              <div
                className="MuiInput-root-86 MuiInput-formControl-87 MuiInput-underline-90"
              >
                <input
                  aria-invalid={false}
                  aria-required={true}
                  autoComplete="username"
                  className="MuiInput-input-94"
                  disabled={false}
                  name="username"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="MuiFormControl-root-72 LocalUserLogin-field-70"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <label
                className="MuiFormLabel-root-81 MuiInputLabel-root-76 MuiInputLabel-formControl-77 MuiInputLabel-animated-80"
                data-shrink={false}
              >
                Password
                <span
                  className="MuiFormLabel-asterisk-85"
                >
                   *
                </span>
              </label>
              <div
                className="MuiInput-root-86 MuiInput-formControl-87 MuiInput-underline-90"
              >
                <input
                  aria-invalid={false}
                  aria-required={true}
                  autoComplete="current-password"
                  className="MuiInput-input-94 MuiInput-inputType-97"
                  disabled={false}
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              className="LocalUserLogin-actions-71"
            >
              <button
                className="MuiButtonBase-root-52 MuiButtonBase-disabled-53 MuiButton-root-99 MuiButton-raised-105 MuiButton-raisedPrimary-106 MuiButton-disabled-109 MuiButton-fullWidth-114"
                disabled={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="-1"
                type="submit"
              >
                <span
                  className="MuiButton-label-100"
                >
                  Log in
                </span>
              </button>
            </div>
          </form>
        </div>
        <div
          className="Login-or-67"
        >
          or
        </div>
        <div
          className="ProviderChoices-root-115"
        >
          <button
            className="MuiButtonBase-root-52 MuiButton-root-99 ProviderButton-root-117 MuiButton-raised-105 MuiButton-raisedPrimary-106"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-100"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-118 ProviderButton-leftIcon-116"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5,3H19C20.1,3 21,3.9 21,5V19C21,20.1 20.1,21 19,21H5C3.9,21 3,20.1 3,19V5C3,3.9 3.9,3 5,3M18,5H15.5C13.57,5 12,6.57 12,8.5V11H10V14H12V21H15V14H18V11H15V9C15,8.45 15.45,8 16,8H18V5Z"
                />
              </svg>
              Log in with Facebook
            </span>
            <span
              className="MuiTouchRipple-root-58"
            />
          </button>
          <button
            className="MuiButtonBase-root-52 MuiButton-root-99 ProviderButton-root-125 MuiButton-raised-105 MuiButton-raisedPrimary-106"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-100"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-118 ProviderButton-leftIcon-124"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5,3H19C20.1,3 21,3.9 21,5V19C21,20.1 20.1,21 19,21H14.56C14.24,20.93 14.23,20.32 14.23,20.11L14.24,17.64C14.24,16.8 13.95,16.25 13.63,15.97C15.64,15.75 17.74,15 17.74,11.53C17.74,10.55 17.39,9.74 16.82,9.11C16.91,8.89 17.22,7.97 16.73,6.73C16.73,6.73 15.97,6.5 14.25,7.66C13.53,7.46 12.77,7.36 12,7.35C11.24,7.36 10.46,7.46 9.75,7.66C8.03,6.5 7.27,6.73 7.27,6.73C6.78,7.97 7.09,8.89 7.18,9.11C6.61,9.74 6.26,10.55 6.26,11.53C6.26,15 8.36,15.75 10.36,16C10.1,16.2 9.87,16.6 9.79,17.18C9.27,17.41 7.97,17.81 7.17,16.43C7.17,16.43 6.69,15.57 5.79,15.5C5.79,15.5 4.91,15.5 5.73,16.05C5.73,16.05 6.32,16.33 6.73,17.37C6.73,17.37 7.25,19.12 9.76,18.58L9.77,20.11C9.77,20.32 9.75,20.93 9.43,21H5C3.9,21 3,20.1 3,19V5C3,3.9 3.9,3 5,3Z"
                />
              </svg>
              Log in with Github
            </span>
            <span
              className="MuiTouchRipple-root-58"
            />
          </button>
          <button
            className="MuiButtonBase-root-52 MuiButton-root-99 ProviderButton-root-127 MuiButton-raised-105 MuiButton-raisedPrimary-106"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-100"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-118 ProviderButton-leftIcon-126"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5,3H19C20.1,3 21,3.9 21,5V19C21,20.1 20.1,21 19,21H5C3.9,21 3,20.1 3,19V5C3,3.9 3.9,3 5,3M17.71,9.33C18.19,8.93 18.75,8.45 19,7.92C18.59,8.13 18.1,8.26 17.56,8.33C18.06,7.97 18.47,7.5 18.68,6.86C18.16,7.14 17.63,7.38 16.97,7.5C15.42,5.63 11.71,7.15 12.37,9.95C9.76,9.79 8.17,8.61 6.85,7.16C6.1,8.38 6.75,10.23 7.64,10.74C7.18,10.71 6.83,10.57 6.5,10.41C6.54,11.95 7.39,12.69 8.58,13.09C8.22,13.16 7.82,13.18 7.44,13.12C7.81,14.19 8.58,14.86 9.9,15C9,15.76 7.34,16.29 6,16.08C7.15,16.81 8.46,17.39 10.28,17.31C14.69,17.11 17.64,13.95 17.71,9.33Z"
                />
              </svg>
              Log in with Twitter
            </span>
            <span
              className="MuiTouchRipple-root-58"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
